x-pack/legacy/plugins/siem/public/pages/detection_engine/rules/components/all_rules_tables/index.tsx

```diff
@@ -4,30 +4,52 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import { EuiBasicTable, EuiTab, EuiTabs, EuiEmptyPrompt } from '@elastic/eui';
+import {
+  EuiBasicTable,
+  EuiBasicTableColumn,
+  EuiTab,
+  EuiTabs,
+  EuiEmptyPrompt,
+  Direction,
+  EuiTableSelectionType,
+} from '@elastic/eui';
 import React, { useMemo, memo, useState } from 'react';
 import styled from 'styled-components';
 
+import { EuiBasicTableOnChange } from '../../types';
 import * as i18n from '../../translations';
-import { RuleStatusRowItemType } from '../../../../../pages/detection_engine/rules/all/columns';
-import { Rules } from '../../../../../containers/detection_engine/rules';
+import {
+  RulesColumns,
+  RuleStatusRowItemType,
+} from '../../../../../pages/detection_engine/rules/all/columns';
+import { Rule, Rules } from '../../../../../containers/detection_engine/rules';
 
 // EuiBasicTable give me a hardtime with adding the ref attributes so I went the easy way
 // after few hours of fight with typescript !!!! I lost :(
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
 const MyEuiBasicTable = styled(EuiBasicTable as any)`` as any;
 
 interface AllRulesTablesProps {
-  euiBasicTableSelectionProps: unknown;
+  euiBasicTableSelectionProps: EuiTableSelectionType<Rule>;
   hasNoPermissions: boolean;
-  monitoringColumns: unknown;
-  paginationMemo: unknown;
+  monitoringColumns: Array<EuiBasicTableColumn<RuleStatusRowItemType>>;
+  pagination: {
+    pageIndex: number;
+    pageSize: number;
+    totalItemCount: number;
+    pageSizeOptions: number[];
+  };
   rules: Rules;
-  rulesColumns: unknown;
-  rulesStatuses: RuleStatusRowItemType[] | null;
-  sorting: unknown;
-  tableOnChangeCallback: unknown;
-  tableRef?: unknown;
+  rulesColumns: RulesColumns[];
+  rulesStatuses: RuleStatusRowItemType[];
+  sorting: {
+    sort: {
+      field: 'enabled';
```

Comment:
I saw in cases they used type `EuiTableSortingType` but then again I know we're limiting it to only sort the `enabled` field. Maybe something like `EuiTableSortingType<Pick<Rule, 'enabled'>>` (that might be off).

Reply:
Since `EuiTableSortingType` has `sorting` as optional, @XavierM mentioned that it might be better to explicitly type this field out, rather than using the eui type.